lib/internal/Magento/Framework/HTTP/Client/Curl.php

```diff
@@ -453,11 +453,11 @@ protected function parseHeaders($ch, $data)
             }
 
             if (strlen($name)) {
-                if ("Set-Cookie" == $name) {
-                    if (!isset($this->_responseHeaders[$name])) {
-                        $this->_responseHeaders[$name] = [];
+                if ('set-cookie' === strtolower($name)) {
+                    if (!isset($this->_responseHeaders['Set-Cookie'])) {
+                        $this->_responseHeaders['Set-Cookie'] = [];
```

Comment:
we can remove this lines.
is sufficient:
`$this->_responseHeaders['Set-Cookie'][] = $value;

Reply:
No. We need to initialize `$this->_responseHeaders['Set-Cookie']` with an empty array if it doesn't exists.

Reply:
@ihor-sviziev it's not necessary.
https://3v4l.org/eSFXF

Reply:
@rvitaliy it is indeed not necessary to initialize an array in PHP. I did not optimize the original code, just fixed the bug and added a test.

PS Do you need me to check code styling / PHPCS / PHPMD or has it already been smoothed out?

Reply:
@pmzandbergen yes, please review and fix static & unit test failures.

Thank you!

Reply:
@ihor-sviziev I'll fix it tomorrow (probably)

Reply:
In `Test/Unit/Client/_files/curl_exec_mock.php` we override function `curl_exec()`, it's good for test `CurlTest`
but the next test `Magento\Framework\HTTP\Test\Unit\Client\CurlTest.testInvalidProtocol` needs to execute original `curl_exec()` function but it still use overrided function `curl_exec()` so that's why test falls.

Reply:
> In `Test/Unit/Client/_files/curl_exec_mock.php` we override function `curl_exec()`, it's good for test `CurlTest`
> but the next test `Magento\Framework\HTTP\Test\Unit\Client\CurlTest.testInvalidProtocol` needs to execute original `curl_exec()` function but it still use overrided function `curl_exec()` so that's why test falls.

Indeed, running both tests in sequence fails. Shouldn't the test environment be cleaned up after running the first test? The `Magento\Framework\HTTP\Test\Unit\Adapter\CurlTest` overrides curl_exec as well and doesn't seem to give any trouble.

Tests have been moved to another folder.